OpenRA.Game/Network/LocalizedMessage.cs

```diff
@@ -75,30 +76,41 @@ static object LoadArguments(MiniYaml yaml)
 			return arguments.ToArray();
 		}
 
-		static readonly string[] SerializeFields = { "Key" };
-
-		public LocalizedMessage(MiniYaml yaml)
+		public LocalizedMessage(ModData modData, MiniYaml yaml)
 		{
+			// Let the FieldLoader do the dirty work of loading the public fields.
 			FieldLoader.Load(this, yaml);
-		}
 
-		public LocalizedMessage(string key, Dictionary<string, object> arguments = null)
-		{
-			Key = key;
-			Arguments = arguments?.Select(a => new FluentArgument(a.Key, a.Value)).ToArray();
+			var argumentDictionary = new Dictionary<string, object>();
+			foreach (var argument in Arguments)
+			{
+				if (argument.Type == FluentArgument.FluentArgumentType.Number)
+				{
+					if (!int.TryParse(argument.Value, out var number))
```

Comment:
`FluentNumber` wraps a double, so this should probably be using `double.TryParse`(?)